src/util/convertHTMLtoJSON.ts

```diff
@@ -0,0 +1,161 @@
+import _ from 'lodash'
+import { Element, HimalayaNode, Text, parse } from 'himalaya'
+import { Block } from '../action-creators/importText'
+
+interface PreBlock extends Block {
+  wasSpan?: boolean,
+  hasSpaces?: boolean,
```

Comment:
Could you explain what `PreBlock` is for me, and the use of `wasSpan` and `hasSpaces`? They seem like very low-level, parsing details, so I'm curious why an intermediate structure must be created for them. Thanks!

Reply:
I have simplified it, now it uses only `partOfThought`. It needs for joining formatting tags.
Example: 
For such HTML:
`<b>one</b> and <i>two</i>`
Himalaya output is
```
[
  {
    "type": "element",
    "tagName": "b",
    "attributes": [],
    "children": [
      {
        "type": "text",
        "content": "one"
      }
    ]
  },
  {
    "type": "text",
    "content": " and "
  },
  {
    "type": "element",
    "tagName": "i",
    "attributes": [],
    "children": [
      {
        "type": "text",
        "content": "two"
      }
    ]
  }
]
```
Result of conversion to PreBlock:
```
[
  {
    scope: "<b>one<b/>",
    children: null,
    partOfThought: true,
  },
  {
    scope: " and ",
    children: null,
  },
  {
    scope: "<i>two<i>",
    children: null,
    partOfThought: true,
  }
]
```
As i mentioned above, PreBlock elements (**not** array of PreBlock) are determined as siblings, but in current case we need to merge such elements together, that's why i use `partOfThought`

Reply:
I see. Thank you for the explanation. Maybe we can combine these in an earlier step to avoid the additional complexity of an intermediate step.